server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -168,6 +170,13 @@ public void parse(ParseContext context) throws IOException {
         multiFields.parse(this, context);
     }
 
+    /**
+     * Returns an index time script for this field mapper, or {@code null} if none is defined
+     */
+    public IndexTimeScript getIndexTimeScript() {
```

Comment:
Yeah, the tricky part here is that MapperScripts can be executed both at index time and at fetch time (for the value fetcher) and those two contexts have entirely different parameter requirements and so different signatures.  I think I can get rid of OneTimeFieldExecutor entirely and just have an anonymous consumer within DocumentParser, and I may be able to unify these two as well by having a class with two methods one it.

Reply:
yea I figured why MapperScript is there, and maybe that should be the only abstraction of a script for this functionality. I don't see the value of the indirection that IndexTimeScript causes now, as it could be just a method exposed directly by the mapper. Maybe you can give that a try and see what happens.

Reply:
We need to have some way of knowing whether or not a FieldMapper has a script on it, so that we can correctly intercept calls to those fields from the index-time SearchLookup and allow scripted fields to refer to each other.  The advantage of having this extra indirection is that it supplies this directly - if the IndexTimeScript is null, then the field doesn't have one.  If we move this to FieldMapper#executeScript then we need to add an additional `hasIndexTimeScript` boolean, which seems like the wrong trade-off to me?

Reply:
I would make FieldMapper script aware, more mappers will support it anyways, and FieldMapper already has getIndexTimeScript anyways for that purpose, plus executing the script itself. Would that work?

Reply:
OK, I have updated things:
* IndexTimeScript is gone
* FieldMapper has two methods, `hasIndexTimeScript` and `executeIndexTimeScript`